### PR TITLE
Feat/shared files table

### PR DESCRIPTION
**Pull request by edge33**
### Ticket:

[User Story 18796](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/18796): share file for view

### Description

- Added table to display users accessing a file

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/upload-api/src/routes/metadata/share/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          200: getGenericResponseSchema(Type.Any()),
+          "4xx": HttpError,
+          "5xx": HttpError,
+        },
+      },
+    },
+    async (request, reply) => {
+      const { fileId, userId } = request.body;
+
+      // TODO: IMPLEMENT THIS
```
Comment (alfonsograziano) [inline apps/upload-api/src/routes/metadata/share/index.ts:38]: Is this a leftover or just a placeholder for a future PR?

File: apps/upload/app/[locale]/(public)/file/[fileId]/actions/shareFile.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"use server";
+
+import { revalidatePath } from "next/cache";
+import { AuthenticationFactory } from "../../../../../utils/authentication-factory";
+import { getCommonLogger } from "nextjs-logging-wrapper";
+
+export default async (fileId: string, prevState, formData) => {
+  const userId = formData.get("userId");
+  console.log({ fileId, userId });
```
Comment (alfonsograziano) [inline apps/upload/app/[locale]/(public)/file/[fileId]/actions/shareFile.ts:9]: CL

File: apps/upload/app/[locale]/(public)/file/[fileId]/components/SharingTables.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+"use client";
```
Comment (alfonsograziano) [inline apps/upload/app/[locale]/(public)/file/[fileId]/components/SharingTables.tsx:1]: Do we need this to be a client component?

**Review by alfonsograziano — DISMISSED** [review]
LGTM - just left a couple of questions :)

File: apps/upload-api/src/routes/metadata/share/index.ts (hunk 1 above)
Comment (edge33) [reply apps/upload-api/src/routes/metadata/share/index.ts:38]: placeholder for a second pr

File: apps/upload/app/[locale]/(public)/file/[fileId]/components/SharingTables.tsx (hunk 3 above)
Comment (edge33) [reply apps/upload/app/[locale]/(public)/file/[fileId]/components/SharingTables.tsx:1]: we do, I have client state in another PR

File: apps/upload-api/src/routes/metadata/share/index.ts (hunk 1 above)
Comment (alfonsograziano) [reply apps/upload-api/src/routes/metadata/share/index.ts:38]: Perfect :)

File: apps/upload/app/[locale]/(public)/file/[fileId]/components/SharingTables.tsx (hunk 3 above)
Comment (alfonsograziano) [reply apps/upload/app/[locale]/(public)/file/[fileId]/components/SharingTables.tsx:1]: Makes sense, thanks!

**Review by alfonsograziano — APPROVED** [review]

**Pull request merged**